### PR TITLE
Drop pkgIds argument from Engine.validatePackages

### PR DESCRIPTION
**Pull request by cocreature**
Doesn’t do anything useful, we just use it to validate that it matches
the keys of pkgs.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala
Comment (cocreature) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala:194]: This is slightly worse in terms of performance. However, I think this is a cost worth paying for the decrease in complexity and the extra safety (much harder to skip package validation accidentally) for two reasons:
1. We really only pay decoding costs/validation here. The package is still not stored to the ledger. Deduplication happens later.
2. Package uploads are extremely rare. We are talking about days if not weeks between uploads so this is absolutely not performance critical.

**Review by remyhaemmerle-da — COMMENTED** [review]
Thanks.

**Review by fabiotudone-da — APPROVED** [review]
LGTM but I'd wait for at least another KV review before merging, if you're not in a hurry.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala:194]: and it actually fix a bug catched by the test I add in 61da6c6c113f32f3c7aaa86eada5dfdd44e8256a

**Review by SamirTalwar-DA — APPROVED** [review]
Seems sensible; no complaints. Thanks for catching the bug!

**Review by miklos-da — APPROVED** [review]

**Pull request merged**
